### PR TITLE
Fork log server framework as map server outline

### PR DESCRIPTION
**Pull request by AlCutter**
Some of this code might look familiar :)
Should probably see if we can factor it out once the dust has settled.

### CODE REVIEW

File: server/vmap/trillian_map_server.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+var (
+	// TODO: Delete when implementation done
+	// ErrNotImplemented is returned when an operation is not supported yet
+	ErrNotImplemented = errors.New("Not yet implemented")
+)
+
+// TODO: There is no access control in the server yet and clients could easily modify
+// any tree.
+
+// LogStorageProviderFunc decouples the server from storage implementations
```
Comment (Martin2112) [inline server/vmap/trillian_map_server.go]: Comment mismatch.

File: server/vmap/trillian_map_server/main.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+var mysqlUriFlag = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test",
+	"uri to use with mysql storage")
+var serverPortFlag = flag.Int("port", 8091, "Port to serve log requests on")
+
+// TODO(Martin2112): Single private key doesn't really work for multi tenant and we can't use
+// an HSM interface in this way. Deferring these issues for later.
+var privateKeyFile = flag.String("private_key_file", "", "File containing a PEM encoded private key")
+var privateKeyPassword = flag.String("private_key_password", "", "Password for server private key")
+
+// TODO(Martin2112): Needs a more realistic provider of log storage with some caching
```
Comment (Martin2112) [inline server/vmap/trillian_map_server/main.go]: Comment still mentions log.

File: server/vmap/trillian_map_server.go (hunk 1 above)
Comment (AlCutter) [inline server/vmap/trillian_map_server.go]: D'oh, updated.

File: server/vmap/trillian_map_server/main.go (hunk 2 above)
Comment (AlCutter) [inline server/vmap/trillian_map_server/main.go]: gnash, updated too.

**Comment by Martin2112** [comment]
OK for now, happy to leave the code sharing until later as there's more important stuff to do.

**Pull request merged**